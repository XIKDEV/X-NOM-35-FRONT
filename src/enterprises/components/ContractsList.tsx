import { FC } from 'react';
import { Card, List } from 'antd';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFileArrowDown } from '@fortawesome/free-solid-svg-icons';
import { IContractsListProps } from '../../interfaces';
import {
  bodyContractStyle,
  listContractsStyleDesktop,
  listContractsStyleMobile,
  listItemContractsStyle,
} from '../ui';

export const ContractsList: FC<IContractsListProps> = ({ data, isMobile }) => {
  return (
    <List
      style={isMobile ? listContractsStyleMobile : listContractsStyleDesktop}
      dataSource={data}
      renderItem={(item) => (
        <List.Item style={listItemContractsStyle}>
          <Card bodyStyle={bodyContractStyle} className="enterprise-contract">
            <div className="enteprise-contract-container flex-row-center space-between">
              <p>{item.contract}</p>
              <p>{item.date}</p>
              <button>
                <FontAwesomeIcon icon={faFileArrowDown} />
              </button>
            </div>
          </Card>
        </List.Item>
      )}
    />
  );
};
