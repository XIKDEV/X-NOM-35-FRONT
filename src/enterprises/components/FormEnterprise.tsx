import { Form } from 'antd';
import {
  FormItemInput,
  FormItemSelectCoordinate,
  FormItemUpload,
  FormsButtons,
} from '../../ui/components';
import { useFormEnterprise } from '../hooks';

export const FormEnterprise = () => {
  const { modules, handleSubmit } = useFormEnterprise();

  return (
    // TODO: Optimizar reglas de formulario
    <Form name="userForm" initialValues={{}} onFinish={handleSubmit}>
      <FormItemInput name="business_name" placeholder="Nombre de la empresa" />
      <FormItemInput name="comercial_name" placeholder="Nombre comercial" />
      <FormItemInput
        name="legal_representative"
        placeholder="Representante legal"
        rules={[
          {
            pattern:
              /^[a-zA-ZÀ-ÿ\u00f1\u00d1]+(\s*[a-zA-ZÀ-ÿ\u00f1\u00d1]*)*[a-zA-ZÀ-ÿ\u00f1\u00d1]+$/,
            message: 'Ingrese nombre válido',
          },
        ]}
      />
      <FormItemInput
        name="RFC"
        placeholder="RFC"
        rules={[
          {
            pattern:
              /^([A-ZÑ&]{3,4}) ?(?:- ?)?(\d{2}(?:0[1-9]|1[0-2])(?:0[1-9]|[12]\d|3[01])) ?(?:- ?)?([A-Z\d]{2})([A\d])$/,
            message: 'Ingrese RFC válido',
          },
        ]}
      />
      <FormItemInput name="street" placeholder="Calle" />
      <div className="two-columns-forms">
        <FormItemInput
          name="exterior_number"
          placeholder="Número exterior"
          rules={[{ pattern: /^[0-9]+/, message: 'Ingrese número válido' }]}
        />
        <FormItemInput name="interior_number" placeholder="Número interior" />
      </div>
      <div className="two-columns-forms">
        <FormItemInput name="suburb" placeholder="Colonia" />
        <FormItemInput
          name="postal_code"
          placeholder="Código postal"
          rules={[{ pattern: /^[0-9]+/, message: 'Ingrese número válido' }]}
        />
      </div>
      <div className="two-columns-forms">
        <FormItemSelectCoordinate
          firstData={modules.states}
          firstName="id_state"
          secondName="id_city"
          firstPlaceholder="Estado"
          secondPlaceholder="Ciudad"
        />
      </div>
      <FormItemInput name="country" placeholder="País" />
      <FormItemInput name="enterprise_type" placeholder="Tipo de empresa" />
      <FormItemInput name="turn_enterprise" placeholder="Giro de la empresa" />
      <FormItemInput
        name="tellphone"
        placeholder="Teléfono"
        rules={[
          { max: 10, message: 'Ingrese teléfono válido' },
          { pattern: /^[0-9]+/, message: 'Ingrese teléfono válido' },
        ]}
      />
      <FormItemInput
        name="email"
        placeholder="Email"
        rules={[{ type: 'email', message: 'Ingrese email válido' }]}
      />

      <FormItemUpload
        name="file"
        accept=".png,.jpeg,.jpg"
        buttonText="Subir imagen"
      />

      <Form.Item>
        <FormsButtons />
      </Form.Item>
    </Form>
  );
};
