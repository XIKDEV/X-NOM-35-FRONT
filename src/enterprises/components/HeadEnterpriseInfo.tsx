import { FC } from 'react';
import { Layout } from 'antd';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPencil } from '@fortawesome/free-solid-svg-icons';
import { IHeadEnterpriseInfoProps } from '../../interfaces';

const { Header } = Layout;

export const HeadEnterpriseInfo: FC<IHeadEnterpriseInfoProps> = ({
  enterpriseName,
}) => {
  return (
    <Header className="header-enterprise">
      <img src="../../../public/XIK_VerdeTransparente.png" alt="" />
      <section className="header-title">
        <h1>{enterpriseName}</h1>
        <button>
          <FontAwesomeIcon icon={faPencil} className="edit-bttn" />
        </button>
      </section>
    </Header>
  );
};
