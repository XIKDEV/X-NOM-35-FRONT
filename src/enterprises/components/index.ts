export * from './ContractsList';
export * from './EnterprisesCard';
export * from './HeadEnterpriseInfo';
export * from './InfoEnterprise';
