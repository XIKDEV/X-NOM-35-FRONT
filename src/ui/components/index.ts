export * from './CardMobile';
export * from './FormItemInput';
export * from './FormItemInputPassword';
export * from './FormItemSelect';
export * from './FormItemSelectCoordinate';
export * from './FormItemUpload';
export * from './FormsButtons';
export * from './FormsDrawer';
export * from './HeaderMobile';
export * from './HeaderPage';
export * from './InfoDrawerButtons';
