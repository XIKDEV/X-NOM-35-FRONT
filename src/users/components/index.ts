export * from './HeaderUserInfo';
export * from './InfoUser';
export * from './UsersCard';
