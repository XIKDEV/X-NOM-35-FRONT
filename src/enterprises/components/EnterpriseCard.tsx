import { FC } from 'react';
import { Card } from 'antd';
import Meta from 'antd/es/card/Meta';
import { ICardProps } from '../../interfaces';
import { enterprisesConstants } from '../../constants';
import { enterprisesCardBodyStyle, enterprisesMetaStyle } from '../ui';

export const EnterpriseCard: FC<ICardProps> = ({
  enterpriseName,
  responsible,
  logoEnterprise,
}) => {
  return (
    <Card
      className="card-enterprise br-8"
      hoverable
      bodyStyle={enterprisesCardBodyStyle}
      cover={
        <img
          alt={`${enterprisesConstants.logoFrom} ${enterpriseName}`}
          src={logoEnterprise}
        />
      }
    >
      <Meta
        title={
          <p className="card-labels" style={enterprisesMetaStyle.title}>
            {enterpriseName}
          </p>
        }
        description={
          <p className="card-labels" style={enterprisesMetaStyle.description}>
            {enterprisesConstants.responsibleEnterprise} {responsible}
          </p>
        }
      />
    </Card>
  );
};
