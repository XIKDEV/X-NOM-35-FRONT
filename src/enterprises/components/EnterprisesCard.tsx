import { FC } from 'react';
import { Card } from 'antd';
import Meta from 'antd/es/card/Meta';
import { ICardProps } from '../../interfaces';
import { cardMetaStyle } from '../../ui/styles';
import { enterprisesCardBodyStyle } from '../ui';

export const EnterprisesCard: FC<ICardProps> = ({
  srcImage,
  title,
  subtitle,
  alt,
  handleOpenDrawerInfo,
}) => {
  return (
    <Card
      className="card-enterprise br-8"
      hoverable
      bodyStyle={enterprisesCardBodyStyle}
      cover={<img src={srcImage} alt={alt} />}
      onClick={handleOpenDrawerInfo}
    >
      <Meta
        title={
          <p className="card-labels" style={cardMetaStyle.title}>
            {title}
          </p>
        }
        description={
          <p className="card-labels" style={cardMetaStyle.description}>
            {subtitle}
          </p>
        }
      />
    </Card>
  );
};
