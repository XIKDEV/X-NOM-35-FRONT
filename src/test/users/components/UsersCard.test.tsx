import { render } from '@testing-library/react';
import { UsersCard } from '../../../users/components';

jest.mock('antd/es/card/Meta', () => {
  return {
    __esModule: true,
    default: () => <div>Mocked Meta Component</div>,
  };
});

describe('Test in <UsersCard />', () => {
  test('should be match with the snapshot', () => {
    const props = {
      srcImage: '../../../../public/XIK_VerdeTransparente.png',
      alt: 'Axel Coronado',
      title: 'Axel Coronado',
      subtitle: 'Responsable de XikDev',
    };
    const { asFragment } = render(<UsersCard {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
