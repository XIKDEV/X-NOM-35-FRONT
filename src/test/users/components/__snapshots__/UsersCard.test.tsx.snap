// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <UsersCard /> should be match with the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered card-user br-4 css-dev-only-do-not-override-17a39f8"
  >
    <div
      class="ant-card-body"
      style="padding: 0px; height: 100%; display: flex; align-items: center;"
    >
      <img
        alt="Axel Coronado"
        class="br-4"
        src="../../../../public/XIK_VerdeTransparente.png"
      />
      <div>
        Mocked Meta Component
      </div>
    </div>
  </div>
</DocumentFragment>
`;
